pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,68 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def _mps_to_right_canonical_representation(mps, max_bond_dim):
+    """
+    Transform a MPS into a right-canonical MPS.
+
+    Args:
+      mps (list[Array]): List of tensors representing the MPS.
+      max_bond_dim (int): Maximum allowed bond dimension (for truncation during SVD).
+
+    Returns:
+      A list of tensors representing the MPS in right-canonical form.
+    """
+
+    L = len(mps)
+    output_mps = [None] * L
+
+    # Procedure analogous to the left-canonical conversion but starting from the right and storing the Vd
+    for i in range(L - 1, 0, -1):
+        chi_left, d, chi_right = mps[i].shape
+        M = mps[i].reshape(chi_left, d * chi_right)
+        U, S, Vd = qml.math.linalg.svd(M, full_matrices=False)
+
+        # Truncate SVD components if needed
+        chi_new = min(int(max_bond_dim), len(S))
+        U = U[:, :chi_new]
+        S = S[:chi_new]
+        Vd = Vd[:chi_new, :]
+
+        output_mps[i] = Vd.reshape(chi_new, d, chi_right)
+
+        US = U @ qml.math.diag(S)
+        mps[i - 1] = qml.math.tensordot(mps[i - 1], US, axes=([2], [0]))
```

Comment:
Is there some kind of reference that explains what `M` `U` S` etc.. refer to?

Reply:
U and S are part of the svd decomposition.
And M is an auxiliary matrix that we use in the process.
I can't find a really useful reference 😕

Reply:
I think [wiki](https://en.wikipedia.org/wiki/Singular_value_decomposition#) does a nice job in explaining U, S and V in terms of rotations and scaling of a matrix `M`, which is the auxiliary matrix as specified by @KetpuntoG above. We could write the workings of this loop in a couple of lines as inline comments.